src/main/java/com/google/sps/servlets/LoadInterviewsServlet.java

```diff
@@ -0,0 +1,276 @@
+import com.google.gson.Gson;
+import com.google.sps.data.Availability;
+import com.google.sps.data.AvailabilityDao;
+import com.google.sps.data.DatastoreAvailabilityDao;
+import com.google.sps.data.DatastoreScheduledInterviewDao;
+import com.google.sps.data.PossibleInterviewSlot;
+import com.google.sps.data.ScheduledInterview;
+import com.google.sps.data.ScheduledInterviewDao;
+import com.google.sps.data.TimeRange;
+import java.io.IOException;
+import java.io.BufferedReader;
+import java.lang.Integer;
+import java.time.Instant;
+import java.time.temporal.ChronoUnit;
+import java.time.ZoneId;
+import java.time.ZoneOffset;
+import java.time.ZonedDateTime;
+import java.time.format.TextStyle;
+import java.util.Locale;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.HashSet;
+import java.util.Set;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import javax.servlet.ServletContext;
+import javax.servlet.ServletException;
+import javax.servlet.RequestDispatcher;
+import java.util.Optional;
+
+@WebServlet("/load-interviews")
+public class LoadInterviewsServlet extends HttpServlet {
+
+  private AvailabilityDao availabilityDao;
+  private ScheduledInterviewDao scheduledInterviewDao;
+  private Instant instant;
+
+  @Override
+  public void init() {
+    init(new DatastoreAvailabilityDao(), new DatastoreScheduledInterviewDao(), Instant.now());
+  }
+
+  public void init(
+      AvailabilityDao availabilityDao,
+      ScheduledInterviewDao scheduledInterviewDao,
+      Instant instant) {
+    this.availabilityDao = availabilityDao;
+    this.scheduledInterviewDao = scheduledInterviewDao;
+    this.instant = instant;
+  }
+
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    int timezoneOffsetMinutes = Integer.parseInt(request.getParameter("timeZoneOffset"));
+    Preconditions.checkArgument(
+        Math.abs(timezoneOffsetMinutes) <= 720,
+        "Offset greater than 720 minutes (12 hours): %s",
+        timezoneOffsetMinutes);
+    ZoneOffset timezoneOffset = convertIntToOffset(timezoneOffsetMinutes);
+    ZonedDateTime day = generateDay(instant, timezoneOffset);
+    ZonedDateTime utcTime = day.withZoneSameInstant(ZoneOffset.UTC);
+    // TODO: Decide how precise this range should be. Must start after or at now, when should end
+    // be?
+    Instant startOfRange = utcTime.toInstant();
+    Instant endOfRange = utcTime.toInstant().plus(6, ChronoUnit.DAYS);
+    List<Availability> availabilitiesInRange =
+        availabilityDao.getInRangeForAll(startOfRange, endOfRange);
+    List<PossibleInterviewSlot> possibleInterviews =
+        getPossibleInterviewSlots(availabilitiesInRange, startOfRange, endOfRange, timezoneOffset);
+    Set<String> dates = new HashSet<String>();
+    for (PossibleInterviewSlot pi : possibleInterviews) {
```

Comment:
nit: change "pi" to "possibleInterview" for readability

Reply:
Done